pkg/aspect/config/accessLogs.proto

```diff
@@ -21,54 +21,61 @@ option go_package="config";
+//                user_agent: request.user-agent | "Mozilla/5.0"
+//      inputs:
+//          template: "{{.time}} {{.user_agent}}"
 message AccessLogsParams {
-    // Provides a way to identify a collection of related log entries.
-    // Default value: "access_log"
+    // Identifies a collection of related log entries.
     string log_name = 1;
 
-    // A list of names of attributes that will be used to generate
-    // access log entries.
-    repeated string attributes = 2;
+    // Describes how attributes must be evaluated to produce values for a log message.
+    message AccessLog {
+        // AccessLogFormat specifies the format to use for generating textual log
+        // entries for the accessLogger aspect of the mixer.
+        //
+        // Default value: COMMON.
+        enum AccessLogFormat {
+            // Invalid default value.
+            ACCESS_LOG_FORMAT_UNSPECIFIED = 0;
 
-    // AccessLogFormat specifies the format to use for generating textual log
-    // entries for the accessLogger aspect of mixer.
-    //
-    // Default value: COMMON.
-    enum AccessLogFormat {
+            // Refers to the [Common Log Format](https://en.wikipedia.org/wiki/Common_Log_Format).
+            // It is used to generate entries like:
+            // `127.0.0.1 user-identifier frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326`
+            COMMON = 1;
 
-        // Refers to the [Common Log Format](https://en.wikipedia.org/wiki/Common_Log_Format).
-        // It is used to generate entries like:
-        // `127.0.0.1 user-identifier frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326`
-        COMMON = 0;
+            // Refers to the Apache [Combined Log Format](https://httpd.apache.org/docs/1.3/logs.html#combined).
+            // It is used to generate entries like:
+            // `127.0.0.1 user-identifier frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326 "http://www.example.com/start.html" "Mozilla/4.08 [en] (Win98; I ;Nav)"`
+            COMBINED = 2;
 
-        // Refers to the Apache [Combined Log Format](https://httpd.apache.org/docs/1.3/logs.html#combined).
-        // It is used to generate entries like:
-        // `127.0.0.1 user-identifier frank [10/Oct/2000:13:55:36 -0700] "GET /apache_pb.gif HTTP/1.0" 200 2326 "http://www.example.com/start.html" "Mozilla/4.08 [en] (Win98; I ;Nav)"`
-        COMBINED = 1;
+            // Used to enable operator-specified formats.
+            CUSTOM = 99;
+        }
+        AccessLogFormat log_format = 1;
 
-        // Used to enable operator-specified formats. This format should be used
-        // in tandem with `custom_log_template` to provide an alternate format
-        // for access log entries.
-        CUSTOM = 99;
-    }
+        // Only used if log_format is CUSTOM. Links this AccessLog to a LogEntryDescriptor
+        // that describes the log's template.
+        string descriptor_name = 2;
```

Comment:
I'm planning on doing that in a follow-on change. The descriptor has already been updated in the istio/api repo but I haven't bumped our SHA for that dependency in istio/mixer yet.